{
	"documenttitle": {
		"text": "Red Hat Security Advisory: Red Hat support for Spring Boot 2.5.10 update",
		"lang": "en"
	},
	"DocumentType": "Security Advisory",
	"documentpublisher": {
		"type": "Vendor",
		"ContactDetails": "secalert@redhat.com",
		"IssuingAuthority": "Red Hat Product Security"
	},
	"documenttracking": {
		"identification": {
			"ID": "RHSA-2022:1179"
		},
		"Status": "Final",
		"Version": "1",
		"revisionhistory": {
			"revision": {
				"Number": "1",
				"Date": "2022-04-12T19:06:00Z",
				"Description": "Current version"
			}
		},
		"InitialReleaseDate": "2022-04-12T19:06:00Z",
		"CurrentReleaseDate": "2022-04-12T19:06:00Z",
		"generator": {
			"Engine": "Red Hat SDEngine 3.4.5",
			"Date": "2022-04-12T22:55:00Z"
		}
	},
	"documentnotes": {
		"note": [
			{
				"text": "\nAn update is now available for Red Hat OpenShift Application Runtimes.\n\nRed Hat Product Security has rated this update as having a security impact of Important. A Common Vulnerability Scoring System (CVSS) base score, which gives a detailed severity rating, is available for each vulnerability. For more information, see the CVE links in the References section.\n    ",
				"title": "Topic",
				"type": "Summary",
				"ordinal": "1",
				"lang": "en"
			},
			{
				"text": "\nRed Hat support for Spring Boot provides an application platform that reduces the complexity of developing and operating applications (monoliths and microservices) for OpenShift as a containerized platform.\n\nThis release of Red Hat support for Spring Boot 2.5.10 serves as a replacement for Red Hat support for Spring Boot 2.4.9, and includes bug fixes and enhancements. For more information, see the release notes listed in the References section.\n\nSecurity Fix(es):\n\n* undertow: client side invocation timeout raised when calling over HTTP2 (CVE-2021-3859)\n\n* tomcat: Infinite loop while reading an unexpected TLS packet when using OpenSSL JSSE engine (CVE-2021-41079)\n\n* tomcat: OutOfMemoryError caused by HTTP upgrade connection leak could lead to DoS (CVE-2021-42340)\n\n* undertow: HTTP2SourceChannel fails to write final frame under some circumstances may lead to DoS (CVE-2021-3597)\n\n* undertow: potential security issue in flow control over HTTP/2 may lead to DOS (CVE-2021-3629)\n\n* wildfly-elytron: possible timing attack in ScramServer (CVE-2021-3642)\n\n* tomcat: HTTP request smuggling when used with a reverse proxy (CVE-2021-33037)\n\n* resteasy: Error message exposes endpoint class information (CVE-2021-20289)\n\n* tomcat: JNDI realm authentication weakness (CVE-2021-30640)\n\nFor more details about the security issue(s), including the impact, a CVSS score, acknowledgments, and other related information, refer to the CVE page(s) listed in the References section.\n    ",
				"title": "Details",
				"type": "General",
				"ordinal": "2",
				"lang": "en"
			},
			{
				"text": "This content is licensed under the Creative Commons Attribution 4.0 International License (https://creativecommons.org/licenses/by/4.0/). If you distribute this content, or a modified version of it, you must provide attribution to Red Hat Inc. and provide a link to the original.",
				"title": "Terms of Use",
				"type": "Legal Disclaimer",
				"ordinal": "3",
				"lang": "en"
			}
		]
	},
	"documentdistribution": {
		"text": "Copyright Â© 2022 Red Hat, Inc. All rights reserved.",
		"lang": "en"
	},
	"aggregateseverity": {
		"text": "Important",
		"namespace": "https://access.redhat.com/security/updates/classification/"
	},
	"documentreferences": {
		"reference": [
			{
				"type": "Self",
				"URL": "https://access.redhat.com/errata/RHSA-2022:1179",
				"Description": "https://access.redhat.com/errata/RHSA-2022:1179"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/security/updates/classification/#important",
				"Description": "https://access.redhat.com/security/updates/classification/#important"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/jbossnetwork/restricted/listSoftware.html?downloadType=distributions&product=catRhoar.spring.boot&version=2.5.10",
				"Description": "https://access.redhat.com/jbossnetwork/restricted/listSoftware.html?downloadType=distributions&product=catRhoar.spring.boot&version=2.5.10"
			},
			{
				"type": "External",
				"URL": "https://access.redhat.com/documentation/en-us/red_hat_support_for_spring_boot/2.5/html/release_notes_for_spring_boot_2.5/index",
				"Description": "https://access.redhat.com/documentation/en-us/red_hat_support_for_spring_boot/2.5/html/release_notes_for_spring_boot_2.5/index"
			}
		]
	},
	"vulnerability": [
		{
			"ordinal": "1",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "undertow: HTTP2SourceChannel fails to write final frame under some circumstances may lead to DoS",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-06-11T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-3597",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2022:1179"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-3597",
						"Description": "CVE-2021-3597"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1970930",
						"Description": "bz#1970930: CVE-2021-3597 undertow: HTTP2SourceChannel fails to write final frame under some circumstances may lead to DoS"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-02-06T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "2",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "undertow: potential security issue in flow control over HTTP/2 may lead to DOS",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-03-29T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-3629",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2022:1179"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-3629",
						"Description": "CVE-2021-3629"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1977362",
						"Description": "bz#1977362: CVE-2021-3629 undertow: potential security issue in flow control over HTTP/2 may lead to DOS"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-04-06T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "3",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "wildfly-elytron: possible timing attack in ScramServer",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-06-30T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-3642",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2022:1179"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-3642",
						"Description": "CVE-2021-3642"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1981407",
						"Description": "bz#1981407: CVE-2021-3642 wildfly-elytron: possible timing attack in ScramServer"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-06-30T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "4",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "undertow: client side invocation timeout raised when calling over HTTP2",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2022-02-01T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-3859",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2022:1179"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-3859",
						"Description": "CVE-2021-3859"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=2010378",
						"Description": "bz#2010378: CVE-2021-3859 undertow: client side invocation timeout raised when calling over HTTP2"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Important"
				}
			},
			"DiscoveryDate": "2021-09-05T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "5",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "resteasy: Error message exposes endpoint class information",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-03-03T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-20289",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2022:1179"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-20289",
						"Description": "CVE-2021-20289"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1935927",
						"Description": "bz#1935927: CVE-2021-20289 resteasy: Error message exposes endpoint class information"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Low"
				}
			},
			"DiscoveryDate": "2021-03-05T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": "Red Hat would like to thank Dirk Papenberg (NTT DATA Germany) for reporting this issue."
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "6",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "tomcat: JNDI realm authentication weakness",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-07-12T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-30640",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2022:1179"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-30640",
						"Description": "CVE-2021-30640"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1981544",
						"Description": "bz#1981544: CVE-2021-30640 tomcat: JNDI realm authentication weakness"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Low"
				}
			},
			"DiscoveryDate": "2021-07-12T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "7",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "tomcat: HTTP request smuggling when used with a reverse proxy",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-07-12T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-33037",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2022:1179"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-33037",
						"Description": "CVE-2021-33037"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=1981533",
						"Description": "bz#1981533: CVE-2021-33037 tomcat: HTTP request smuggling when used with a reverse proxy"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Moderate"
				}
			},
			"DiscoveryDate": "2021-07-12T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "8",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "tomcat: Infinite loop while reading an unexpected TLS packet when using OpenSSL JSSE engine",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-09-15T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-41079",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2022:1179"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "https://lists.apache.org/thread.html/rccdef0349fdf4fb73a4e4403095446d7fe6264e0a58e2df5c6799434%40%3Cannounce.tomcat.apache.org%3E",
						"Description": "https://lists.apache.org/thread.html/rccdef0349fdf4fb73a4e4403095446d7fe6264e0a58e2df5c6799434%40%3Cannounce.tomcat.apache.org%3E"
					},
					{
						"URL": "https://tomcat.apache.org/security-10.html#Fixed_in_Apache_Tomcat_10.0.4",
						"Description": "https://tomcat.apache.org/security-10.html#Fixed_in_Apache_Tomcat_10.0.4"
					},
					{
						"URL": "https://tomcat.apache.org/security-8.html#Fixed_in_Apache_Tomcat_8.5.64",
						"Description": "https://tomcat.apache.org/security-8.html#Fixed_in_Apache_Tomcat_8.5.64"
					},
					{
						"URL": "https://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.44",
						"Description": "https://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.44"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-41079",
						"Description": "CVE-2021-41079"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=2004820",
						"Description": "bz#2004820: CVE-2021-41079 tomcat: Infinite loop while reading an unexpected TLS packet when using OpenSSL JSSE engine"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Important"
				}
			},
			"DiscoveryDate": "2021-09-15T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		},
		{
			"ordinal": "9",
			"xmlns": "http://www.icasi.org/CVRF/schema/vuln/1.1",
			"notes": {
				"note": {
					"text": "tomcat: OutOfMemoryError caused by HTTP upgrade connection leak could lead to DoS",
					"title": "Vulnerability Description",
					"type": "General",
					"ordinal": "1",
					"lang": "en"
				}
			},
			"ReleaseDate": "2021-10-14T00:00:00Z",
			"involvements": {
				"involvement": {
					"party": "Vendor",
					"status": "Completed"
				}
			},
			"CVE": "CVE-2021-42340",
			"remediations": {
				"remediation": {
					"type": "Vendor Fix",
					"description": {
						"text": "\nBefore applying the update, back up your existing installation, including all applications, configuration files, databases and database settings, and so on.\n\nThe References section of this erratum contains a download link for the update. You must be logged in to download the update.\n    ",
						"lang": "en"
					},
					"URL": "https://access.redhat.com/errata/RHSA-2022:1179"
				}
			},
			"references": {
				"reference": [
					{
						"URL": "http://tomcat.apache.org/security-10.html#Fixed_in_Apache_Tomcat_10.0.12",
						"Description": "http://tomcat.apache.org/security-10.html#Fixed_in_Apache_Tomcat_10.0.12"
					},
					{
						"URL": "http://tomcat.apache.org/security-10.html#Fixed_in_Apache_Tomcat_10.1.0-M6",
						"Description": "http://tomcat.apache.org/security-10.html#Fixed_in_Apache_Tomcat_10.1.0-M6"
					},
					{
						"URL": "http://tomcat.apache.org/security-8.html#Fixed_in_Apache_Tomcat_8.5.72",
						"Description": "http://tomcat.apache.org/security-8.html#Fixed_in_Apache_Tomcat_8.5.72"
					},
					{
						"URL": "http://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.54",
						"Description": "http://tomcat.apache.org/security-9.html#Fixed_in_Apache_Tomcat_9.0.54"
					},
					{
						"URL": "https://lists.apache.org/thread.html/r83a35be60f06aca2065f188ee542b9099695d57ced2e70e0885f905c%40%3Cannounce.tomcat.apache.org%3E",
						"Description": "https://lists.apache.org/thread.html/r83a35be60f06aca2065f188ee542b9099695d57ced2e70e0885f905c%40%3Cannounce.tomcat.apache.org%3E"
					},
					{
						"URL": "https://access.redhat.com/security/cve/CVE-2021-42340",
						"Description": "CVE-2021-42340"
					},
					{
						"URL": "https://bugzilla.redhat.com/show_bug.cgi?id=2014356",
						"Description": "bz#2014356: CVE-2021-42340 tomcat: OutOfMemoryError caused by HTTP upgrade connection leak could lead to DoS"
					}
				]
			},
			"threats": {
				"threat": {
					"type": "Impact",
					"Description": "Important"
				}
			},
			"DiscoveryDate": "2021-10-14T00:00:00Z",
			"productstatuses": {
				"status": {
					"ProductID": null
				}
			},
			"acknowledgments": {
				"acknowledgment": {
					"Description": ""
				}
			},
			"cvssscoresets": {
				"scoreset": {
					"BaseScore": "",
					"Vector": ""
				}
			}
		}
	],
	"producttree": {}
}
